src/ChunkMap.cpp

```diff
@@ -1088,7 +1088,7 @@ int  cChunkMap::GetHeight(int a_BlockX, int a_BlockZ)
 		}
 
 		// The chunk is not valid, wait for it to become valid:
-		cCSUnlock Unlock(Lock);
+		m_CSLayers.Unlock();
 		m_evtChunkValid.Wait();
```

Comment:
No way, now you leave the function (line 1082 or 1087) with the chunkmap still closed. There's a reason we use RAII for locking, and we're not removing the RAII just to be able to use some weird annotations.

If anything, make the RAII Lock local to the needed block (make it go out of scope before reaching the `m_evtChunkValid.Wait()` line)